packages/client/src/runtime/core/errorRendering/applyValidationError.test.ts

```diff
@@ -0,0 +1,1495 @@
+        },
+        {
+          where: { email: 123 },
+        },
+      ),
+    ).toMatchInlineSnapshot(`
+      {
+        where: {
+          email: 123
+                 ~~~
+        }
+      }
+
+      Argument gt: Invalid value provided. Expected String or StringFilter, provided Int.
+    `)
+  })
+})
+
+describe('SomeFieldsMissing', () => {
+  test('simple', () => {
+    expect(
+      renderError(
+        {
+          kind: 'SomeFieldsMissing',
+          selectionPath: [],
+          argumentPath: ['where'],
+          inputType: {
+            kind: 'object',
+            name: 'UserWhereUniqueInput',
+            fields: [
+              { name: 'id', typeNames: ['String'], required: false },
+              { name: 'email', typeNames: ['String'], required: false },
+            ],
+          },
+          constraints: { minFieldCount: 1, requiredFields: null },
+        },
+        { where: {} },
+      ),
+    ).toMatchInlineSnapshot(`
+      {
+        where: {
+      ?   id?: String,
+      ?   email?: String
+        }
+      }
+
+      Argument where of type UserWhereUniqueInput needs at least one argument. Available options are listed in green.
+    `)
+  })
+
+  test('multiple', () => {
+    expect(
+      renderError(
+        {
+          kind: 'SomeFieldsMissing',
+          selectionPath: [],
+          argumentPath: ['where'],
+          inputType: {
+            kind: 'object',
+            name: 'UserWhereUniqueInput',
+            fields: [
+              { name: 'id', typeNames: ['String'], required: false },
+              { name: 'email', typeNames: ['String'], required: false },
+            ],
+          },
+          constraints: { minFieldCount: 2, requiredFields: null },
+        },
+        { where: {} },
+      ),
+    ).toMatchInlineSnapshot(`
+      {
+        where: {
+      ?   id?: String,
+      ?   email?: String
+        }
+      }
+
+      Argument where of type UserWhereUniqueInput needs at least 2 arguments. Available options are listed in green.
+    `)
+  })
+
+  test('nested selection', () => {
+    expect(
+      renderError(
+        {
+          kind: 'SomeFieldsMissing',
+          selectionPath: ['user'],
+          argumentPath: ['where'],
+          inputType: {
+            kind: 'object',
+            name: 'UserWhereUniqueInput',
+            fields: [
+              { name: 'id', typeNames: ['String'], required: false },
+              { name: 'email', typeNames: ['String'], required: false },
+            ],
+          },
+          constraints: { minFieldCount: 1, requiredFields: null },
+        },
+        { include: { user: { where: {} } } },
+      ),
+    ).toMatchInlineSnapshot(`
+      {
+        include: {
+          user: {
+            where: {
+      ?       id?: String,
+      ?       email?: String
+            }
+          }
+        }
+      }
+
+      Argument where of type UserWhereUniqueInput needs at least one argument. Available options are listed in green.
```

Comment:
It's hard to see the colors in the snapshot. Could we replace the ansi codes with eg `<green>...</green>`?

Reply:
Added snapshot serializer that does just that. Downside is, it is harder to judge the alignment of different lines because of added extra content. Let me know which one would you prefer.

Reply:
I guess we could have a test that is colorless and one that is with color?